gui/src/avopenglwidget.cpp

```diff
@@ -256,26 +313,31 @@ void AVOpenGLWidget::initializeGL()
 
 	for(int i=0; i<2; i++)
 	{
-		f->glGenTextures(3, frames[i].tex);
-		f->glGenBuffers(3, frames[i].pbo);
-		uint8_t uv_default = 127;
-		for(int j=0; j<3; j++)
+		frames[i].conversion_config = conversion_config;
+		f->glGenTextures(conversion_config->planes, frames[i].tex);
+		f->glGenBuffers(conversion_config->planes, frames[i].pbo);
+		uint8_t uv_default[] = {0x7f, 0x7f};
+		for(int j=0; j<conversion_config->planes; j++)
 		{
 			f->glBindTexture(GL_TEXTURE_2D, frames[i].tex[j]);
 			f->glTexParameterf(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_LINEAR);
 			f->glTexParameterf(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_LINEAR);
 			f->glTexParameterf(GL_TEXTURE_2D, GL_TEXTURE_WRAP_S, GL_CLAMP_TO_EDGE);
 			f->glTexParameterf(GL_TEXTURE_2D, GL_TEXTURE_WRAP_T, GL_CLAMP_TO_EDGE);
-			f->glTexImage2D(GL_TEXTURE_2D, 0, GL_R8, 1, 1, 0, GL_RED, GL_UNSIGNED_BYTE, j > 0 ? &uv_default : nullptr);
+			f->glTexImage2D(GL_TEXTURE_2D, 0, conversion_config->plane_configs[j].internal_format, 1, 1, 0, conversion_config->plane_configs[j].format, GL_UNSIGNED_BYTE, j > 0 ? &uv_default : nullptr);
```

Comment:
Nice, you found the i/j issue yourself, I was just about to comment about it.
Just a small suggestion, better use the array directly here.
```suggestion
			f->glTexImage2D(GL_TEXTURE_2D, 0, conversion_config->plane_configs[j].internal_format, 1, 1, 0, conversion_config->plane_configs[j].format, GL_UNSIGNED_BYTE, j > 0 ? uv_default : nullptr);
```